[12.0][MIG] sale_blanket_order module

tafaRU opened the pull request.

andreparames committed:
[10.0][ADD] sale_blanket_orders

andreparames committed:
sale_blanket_order: don't copy name nor confirm state

When duplicating a confirmed blanket order, the new copy shouldn't
keep the state nor the sequence number (name).

andreparames committed:
sale_blanket_order: update after review

andreparames committed:
sale_blanket_order: Re-calculate price based on qty

andreparames committed:
sale_blanket_order: make prohibition of adding new lines optional

rousseldenis committed:
[FIX] sale_blanket_order: Ondelete cascade on blanket lines

AdriaGForgeFlow committed:
[11.0][MIG] Migrate module sale_blanket_order

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update sale_blanket_order.pot

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 7.3% (9 of 123 strings)

Translation: sale-workflow-11.0/sale-workflow-11.0-sale_blanket_order
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-11-0/sale-workflow-11-0-sale_blanket_order/de/

OCA-git-bot committed:
[UPD] README.rst

tafaRU committed:
[MIG] sale_blanket_order: Migration to 12.0

tafaRU commented:
@acsone @eficent any of you could please review it? Thank you in advance.

LoisRForgeFlow commented:
@ageficent Could you have a look?

AdriaGForgeFlow commented on sale_blanket_order/wizard/create_sale_orders.py:
```diff
@@ -142,63 +137,53 @@ def create_sale_order(self):
             res.append(sale_order.id)
         return {
             'domain': [('id', 'in', res)],
-            'name': _('Sales Orders'),
```
Why do you remove this? Shouldn't it be translatable?

tafaRU replied:
It already is: https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_actions.py#L173

AdriaGForgeFlow replied:
This is not an 'ir.actions.act_window' record, it is created on the go from python.
AFAIK, if you don't set the name to be translatable here, it will not appear in the pot file for its translation.

tafaRU replied:
fixed

tafaRU replied:
thanks, my fault, I trusted https://github.com/OCA/sale-workflow/pull/888#discussion_r329150525 without investigating deeper :wink:

AdriaGForgeFlow reviewed (commented):
Functional review: Overall works fine, I found one problem when duplicating Sale Blanket Orders the state is kept. If you have a confirmed Blanket Order and you duplicate it the newly created one will have state confirmed and name "Draft" which does not make sense.

Code review: just one comment below:

tafaRU commented:
@ageficent about your functional remark, for consinstency, I could add `default='draft` to the field `state` of the `blanket.order`. What do you think about it?

AdriaGForgeFlow commented:
> @ageficent about your functional remark, for consinstency, I could add `default='draft` to the field `state` of the `blanket.order`. What do you think about it?

That would not prevent the state from being copied. You need to add, in addition, the `copy=False` parameter to the field definition. 
This is an example: https://github.com/OCA/sale-workflow/pull/966/files#diff-b175af03110d728bd289598dd24e4106R49

tafaRU commented:
> That would not prevent the state from being copied. You need to add, in addition, the copy=False parameter to the field definition.

It already is: https://github.com/tafaRU/sale-workflow/blob/12.0-mig-sale_blanket_order/sale_blanket_order/models/blanket_orders.py#L76

What I suggested it is also add `default='draft'`.

AdriaGForgeFlow commented:
> > That would not prevent the state from being copied. You need to add, in addition, the copy=False parameter to the field definition.
> 
> It already is: https://github.com/tafaRU/sale-workflow/blob/12.0-mig-sale_blanket_order/sale_blanket_order/models/blanket_orders.py#L76
> 
> What I suggested it is also add `default='draft'`.

You're right. My bad, I didn't check. Then I guess with the default should do the trick.

tafaRU commented:
@ageficent, just force pushed new changes. Please try again when runbot build will be ready.
Thanks!

AdriaGForgeFlow reviewed (approved):
Looks good now! Thanks for the work :+1:

tafaRU committed:
sale_blanket_order: fix the following use case scenario.

Steps to reproduce:

* create and confirm a blanket order (BO) with (product A, qty 30) and (product B, qty 20)
* from the BO create a SO with (product A, qty 10) and (product B, qty 10)
* from the BO create a SO with (product A, qty 20) and (product B, qty 0)
* from the BO create another SO with (product B, qty 10)

Current behavior:

It raises the exception "The sale has already been completed.".

Expected behavior:

No exception is raised.

tafaRU commented:
@ageficent, just pushed a fix for a bug already existing in the previous versions. Could you please update your review?

Thank you in advance.

AdriaGForgeFlow reviewed (approved):
Code changes look good :+1: 
Thanks!

tafaRU commented:
@ageficent, please also review https://github.com/OCA/sale-workflow/pull/982

stevech091 reviewed (approved):
Functional OK

HviorForgeFlow reviewed (approved):
Functional review for fixed bug :+1:

tafaRU commented:
@OCA/crm-sales-marketing-maintainers could you please merging this?
I'd do it by myself but I don't want to violate the rule about self-merging. BTW, is it always valid?

tafaRU commented:
ping @OCA/crm-sales-marketing-maintainers!

eLBati commented:
Scope question: what's the difference from native blanket orders?
https://www.odoo.com/documentation/user/12.0/purchase/purchases/tender/blanket_orders.html

eLBati commented:
Never mind, this is about sale

eLBati commented on sale_blanket_order/readme/USAGE.rst:
```diff
@@ -0,0 +1,53 @@
+
+To create a new Sale Blanket Order go to the sale menu in the Sales section:
+
+.. image:: /sale_blanket_order/static/description/BO_menu.png
```
Please use suggested syntax
https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/readme/CONFIGURE.rst
to correctly display images in both README and index

tafaRU replied:
Thanks @eLBati, I applied changes as you suggested but it only works in https://github.com/tafaRU/sale-workflow/blob/12.0-mig-sale_blanket_order/sale_blanket_order/readme/USAGE.rst
Trying to regenerate the README file I get https://raw.githubusercontent.com/tafaRU/sale-workflow/12.0-mig-sale_blanket_order/sale_blanket_order/README.rst that doesn't work.
Any idea why that might be?

HviorForgeFlow replied:
Will work when merged

tafaRU replied:
@hveficent, ah cool! I wasn't aware of it. Do you have any reference?

tafaRU committed:
sale_blanket_order: correctly display images in both README and index

eLBati reviewed (approved):
Thanks

eLBati commented:
/ocabot merge
